testsuite/tests/comprehensions/comprehensions.ml

```diff
@@ -532,7 +532,7 @@ List.rev !var;;
 [%%expect{|
 - : unit = ()
 - : int array = [|0; 1; 2; 3; 4; 5; 0; 1; 2; 3; 4; 5|]
-- : int list = [3; 4; 0; 5]
+- : int list = [0; 5; 3; 4]
```

Comment:
That is the wrong order. It should evaluate the outer loop before the inner loop.

Reply:
As mentioned in the PRs description, I think this new order is slightly less surprising although either order would be reasonable. Note that it is only the side-effects within a single "block" (i.e. separated by an `and`) that are going left-to-right, side affects in different blocks go in the order determined by their binding structure (right-to-left).

In general, OCaml says that side-effects either side of an `and` are implementation defined. e.g.:
```ocaml
let () = print_string "1"
and () = print_string "2" in ()
```
is in principle allowed to print `12` or `21`. (Although, it will in practice print `12`, and I happen to know that the compiler will fail to build if you make the side-effects go the other way, so this point is somewhat academic)

Reply:
We should change the order as discussed offline.